qbft/instance.go

```diff
@@ -121,7 +132,7 @@ func (i *Instance) ProcessMsg(msg *SignedMessage) (decided bool, decidedValue []
 			}
 			return err
 		case RoundChangeMsgType:
-			return i.uponRoundChange(i.StartValue, msg, i.State.RoundChangeContainer, i.config.GetValueCheckF())
+			return i.uponRoundChange(msg, i.State.RoundChangeContainer, i.config.GetValueCheckF())
```

Comment:
I noticed we're passing `i.StartValue`, `i.State.RoundChangeContainer`, and `i.config.GetValueCheckF()` as arguments to the function. Given these are accessed via the instance `i`, could someone help clarify the rationale behind explicitly passing these as parameters instead of accessing them directly within the uponRoundChange method?

Is there a specific design choice or benefit we're aiming for by structuring the call this way? For instance, does this approach enhance testability, readability, or separation of concerns in a way that accessing them directly within the function would not?

Reply:
ditto
`i.uponProposal`
`i.uponPrepare`
`i.UponCommit`

Reply:
@GalRogozinski I agree with this, either only pass and not save (if its only used within these function) or save and don't pass.

Reply:
this is out of scope of PR, but can be done